pkg/internal/config/config.go

```diff
@@ -120,64 +120,88 @@ func New(baseConfig kconfig.Config) (Config, error) {
 		}
-		newConfig.TrafficGeneratorEastMacAddress = trafficGeneratorEastMacAddress
 	}
 
-	if rawTrafficGeneratorWestMacAddress :=
-		baseConfig.Params[TrafficGeneratorWestMacAddressParamName]; rawTrafficGeneratorWestMacAddress != "" {
-		trafficGeneratorWestMacAddress, err := net.ParseMAC(rawTrafficGeneratorWestMacAddress)
+	if rawVal := baseConfig.Params[TrafficGeneratorWestMacAddressParamName]; rawVal != "" {
+		newConfig.TrafficGeneratorWestMacAddress, err = parseMacAddress(rawVal, ErrInvalidTrafficGeneratorWestMacAddress)
 		if err != nil {
-			return Config{}, ErrInvalidTrafficGeneratorWestMacAddress
+			return Config{}, err
 		}
-		newConfig.TrafficGeneratorWestMacAddress = trafficGeneratorWestMacAddress
 	}
 
-	if rawEastDPDKMacAddress := baseConfig.Params[DPDKEastMacAddressParamName]; rawEastDPDKMacAddress != "" {
-		dpdkEastMacAddress, err := net.ParseMAC(rawEastDPDKMacAddress)
+	if rawVal := baseConfig.Params[DPDKEastMacAddressParamName]; rawVal != "" {
+		newConfig.DPDKEastMacAddress, err = parseMacAddress(rawVal, ErrInvalidDPDKEastMacAddress)
 		if err != nil {
-			return Config{}, ErrInvalidDPDKEastMacAddress
+			return Config{}, err
 		}
-		newConfig.DPDKEastMacAddress = dpdkEastMacAddress
 	}
 
-	if rawWestDPDKMacAddress := baseConfig.Params[DPDKWestMacAddressParamName]; rawWestDPDKMacAddress != "" {
-		dpdkWestMacAddress, err := net.ParseMAC(rawWestDPDKMacAddress)
+	if rawVal := baseConfig.Params[DPDKWestMacAddressParamName]; rawVal != "" {
+		newConfig.DPDKWestMacAddress, err = parseMacAddress(rawVal, ErrInvalidDPDKWestMacAddress)
 		if err != nil {
-			return Config{}, ErrInvalidDPDKWestMacAddress
+			return Config{}, err
 		}
-		newConfig.DPDKWestMacAddress = dpdkWestMacAddress
 	}
 
-	if rawTestDuration := baseConfig.Params[TestDurationParamName]; rawTestDuration != "" {
-		testDuration, err := time.ParseDuration(rawTestDuration)
+	if rawVal := baseConfig.Params[TestDurationParamName]; rawVal != "" {
+		newConfig.TestDuration, err = parseTestDuration(rawVal)
 		if err != nil {
-			return Config{}, ErrInvalidTestDuration
+			return Config{}, err
 		}
-		newConfig.TestDuration = testDuration
 	}
 
 	return newConfig, nil
 }
+
+func parseTrafficGeneratorPacketsPerSecondInMillions(rawVal string) (int, error) {
+	val, err := strconv.Atoi(rawVal)
+	if err != nil || val < 0 {
+		return 0, ErrInvalidTrafficGeneratorPacketsPerSecondInMillions
+	}
+	return val, nil
+}
+
+func parsePortBandwidthGB(rawVal string) (int, error) {
+	val, err := strconv.Atoi(rawVal)
+	if err != nil || val <= 0 {
+		return 0, ErrInvalidPortBandwidthGB
+	}
+	return val, nil
+}
+
+func parseMacAddress(rawVal string, outputErr error) (net.HardwareAddr, error) {
+	val, err := net.ParseMAC(rawVal)
+	if err != nil {
+		return nil, outputErr
+	}
+	return val, nil
+}
+
+func parseTestDuration(rawVal string) (time.Duration, error) {
+	val, err := time.ParseDuration(rawVal)
+	if err != nil {
+		return time.Duration(0), ErrInvalidTestDuration
+	}
+	return val, nil
+}
```

Comment:
Could you please explain what is the added value in calling these wrapper functions over of directly calling `net.ParseMAC()` and `time.ParseDuration()`?

Reply:
Is "it would cause the linter to pass" considered added value? :)

Reply:
I find `parseTrafficGeneratorPacketsPerSecondInMillions`, `parseMacAddress`, `parseTestDuration` to be redundant - IMO they don't add value over directly calling the function they are wrapping.
Furthermore,  they hide the returned error message.

Please consider this suggestion:
```go
if rawVal := baseConfig.Params[PortBandwidthGBParamName]; rawVal != "" {
    newConfig.PortBandwidthGB, err = parseNonZeroPositiveInt(rawVal)
    if err != nil {
        return Config{}, ErrInvalidPortBandwidthGB
    }
}
```

Reply:
What is the problem with the linter?

Reply:
never mind it's fixed now

Reply:
Done, I took your suggestion and it fixed the linter as well